### PR TITLE
EOS-18493: Create a generic create user api

### PR DESCRIPTION
**Pull request by sradhanand-pati**
Signed-off-by: sradhanandpati <sradhanand.pati@seagate.com>

### CODE REVIEW

File: api/python/provisioner/api.py [hunk 1]
```diff
@@ -578,17 +578,20 @@ def replace_node(node_id, node_host=None, node_port=None, nowait=True):
     )
 
 
-def create_user(uname, passwd, targets=ALL_MINIONS, nowait=False):
+def create_user(uname, passwd, sudo=True, group_list=[], commands_list=[], targets=ALL_MINIONS, nowait=False):
```
Comment (ypise) [inline api/python/provisioner/api.py:581]: Python has a better way for declaring types now:
```
def create_user(
    uname: str,
    passwd: str,
    sudo=True: bool,
    group_list: list,
    commands_list: list,
    targets=ALL_MINIONS,
    nowait=False: bool
):
```

File: api/python/provisioner/commands/create_user.py [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+
+    targets: str = RunArgs.targets
+
+
+@attr.s(auto_attribs=True)
+class CreateUser(CommandParserFillerMixin):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = RunArgsUser
+
+    def run(
```
Comment (ypise) [inline api/python/provisioner/commands/create_user.py:96]: Improvement would help:
```
def run(
    self,
    uname: str,
    passwd: str,
    sudo=True: bool,
    group_list: list,
    commands_list: list,
    targets=ALL_MINIONS,
):
```

File: api/python/provisioner/commands/create_user.py [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+                hostname_pillar = PillarResolver(LOCAL_MINION).get([hostname])
+                hostname_pillar = next(iter(hostname_pillar.values()))
+
+                if (not hostname_pillar[hostname] or
+                        hostname_pillar[hostname] is values.MISSED):
+                    raise BadPillarDataError(
+                        'value for {} is not specified'.format(hostname.pi_key)
+                    )
+
+                ssh_config = f'''Host {node} {hostname_pillar[hostname]}
```
Comment (ypise) [inline api/python/provisioner/commands/create_user.py:189]: We might want to make this optional.
Add a TODO here

File: api/python/provisioner/commands/create_user.py [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+                StateFunExecuter.execute(
+                    'file.append',
+                    fun_kwargs=dict(
+                        name=str(user_fileroots_dir / 'config'),
+                        text=ssh_config
+                    )
+                )
+
+        def _copy_minion_nodes():
+            StateFunExecuter.execute(
```
Comment (ypise) [inline api/python/provisioner/commands/create_user.py:208]: What are we copying here. Add a docstring or comment.

**Review by ypise — APPROVED** [review]
Few suggestions to address.

File: api/python/provisioner/api.py (hunk 1 above)
Comment (sradhanand-pati) [reply api/python/provisioner/api.py]: Addressed

File: api/python/provisioner/commands/create_user.py (hunk 2 above)
Comment (sradhanand-pati) [reply api/python/provisioner/commands/create_user.py:96]: Addressed

File: api/python/provisioner/commands/create_user.py (hunk 4 above)
Comment (sradhanand-pati) [reply api/python/provisioner/commands/create_user.py:212]: Addressed

File: api/python/provisioner/commands/create_user.py (hunk 3 above)
Comment (sradhanand-pati) [reply api/python/provisioner/commands/create_user.py:190]: Addressed

**Review by ypise — APPROVED** [review]

**Review by mazinamdar — APPROVED** [review]

**Pull request merged**